### PR TITLE
FIX: set_url() without effect in the plot for instances of Tick

### PR DESCRIPTION
**Pull request by khyox**
## PR Summary

In the class Tick, the url attribute is not passed to any of the objects that are to be drawn, so it is useless, as detailed in issue #9695. Suggested solution is to override the `set_url()` method in Tick to transfer the url attribute content to the tick labels, that are to be drawn.

Fixes issue #9695.

## PR Checklist

- [N/A] Has Pytest style unit tests
- [OK] Code is PEP 8 compliant
- [N/A] New features are documented, with examples if plot related
- [OK] Documentation is sphinx and numpydoc compliant
- [N/A] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [N/A] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jklymak** [comment]
Why are tests N/A for this?  If it had a test, this funtionality wouldn't break in the future...

File: lib/matplotlib/axis.py [hunk 1]
```diff
@@ -306,6 +306,16 @@ def set_label2(self, s):
         self.label2.set_text(s)
         self.stale = True
 
+    def set_url(self, s):
+        """
+        Set the url of label1 and label2
+
+        ACCEPTS: str
+        """
```
Comment (tacaswell) [inline lib/matplotlib/axis.py]: Can you also call `super(self, Tick).set_label(s)`?

**Comment by tacaswell** [comment]
How can we add a test for this?  Can we parse the XML that is svg well using only our current dependencies to verify that the url is there?

File: lib/matplotlib/axis.py (hunk 1 above)
Comment (khyox) [reply lib/matplotlib/axis.py]: Hi Thomas, if you meant to add a call to `super(Tick, self).set_url(s)`, I just added it, if not, I don't get it, sorry.

Comment (khyox) [reply lib/matplotlib/axis.py]: Travis is failing with that addition with some `ImageComparisonFailure: images not close` errors. When calling the parent Artist's set_url() method it just does `self._url = url`! Amazing.

Comment (tacaswell) [reply lib/matplotlib/axis.py]: That is what I meant, thanks.

Can you reproduce the failures locally?

**Comment by jklymak** [comment]
@tacaswell, I was only thinking a test could be devised to make sure the url was attached to the tick.  Whether its embeded or not is another issue.

**Comment by anntzer** [comment]
For testing one can probably extract the node with the right XPath query (https://docs.python.org/3.6/library/xml.etree.elementtree.html#xpath-support) and do the right assert on it...

**Comment by QuLogic** [comment]
I've incorporated this into #17338 and added a test.

**Comment by khyox** [comment]
Many thanks, @QuLogic, and sorry, it seems that this fell off my radar at some point :(

**Comment by tacaswell** [comment]
@khyox It fell off ours too :disappointed:

**Pull request closed without merging**